OSDOCS-7678:Included missing policy definition and instructions from … (#64483)
openshift/openshift-docs: modules/rosa-sharing-vpc-creation-and-sharing.adoc

.Procedure

. From the AWS account that centrally manages your VPC, create or modify a VPC to your specifications in the link:https://us-east-1.console.aws.amazon.com/vpc/[VPC section of the AWS console]. This AWS account will be the *VPC-owning AWS account*.
+

> . Create a custom policy file with minimal route53 permissions using the name `Route53minimalPermissions`:

Reviewer: ```suggestion
. Create a custom policy file to allow for necessary Shared VPC permissions using the name `SharedVPCPolicy `:
```

+
[source,terminal]
----

> cat <<EOF > /tmp/route53-policy.json

Reviewer: ```suggestion
$ cat <<EOF > /tmp/shared-vpc-policy.json
```

{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "route53:ChangeResourceRecordSets",
                "route53:ListHostedZones",
                "route53:ListHostedZonesByName",
                "route53:ListResourceRecordSets",
                "route53:ChangeTagsForResource",
                "route53:GetAccountLimit",
                "route53:GetChange",
                "route53:GetHostedZone",
                "route53:ListTagsForResource",

>                 "route53:UpdateHostedZoneComment"

Reviewer: ```suggestion
                "route53:UpdateHostedZoneComment",
                "tag:GetResources",
                "tag:UntagResources"
```

Based on work by @jharrington22: https://github.com/openshift/ops-sop/pull/2770/files#diff-4c661b9862858e5b5a7a7100bc6e5d781fcdccf16199ad4009b2d395eae0db20R166-R177

            ],
            "Resource": "*"
        }
    ]
}
EOF
----
+
.. Create the policy in AWS:
+
[source, terminal]
----
$ aws iam create-policy \

>     --policy-name Route53minimalPermissions \

Reviewer: ```suggestion
    --policy-name SharedVPCPolicy \
```

>     --policy-document file:///tmp/route53-policy.json

Reviewer: ```suggestion
    --policy-document file:///tmp/shared-vpc-policy.json
```

----
+
You will attach this policy to a role necessary for the shared VPC permissions.
+

> . Create a custom trust policy file that grants permission to assume roles:

Reviewer: ```suggestion
. Create a Shared VPC role grants permission to assume roles:
```

+
[source,terminal]
----

> cat <<EOF > /tmp/trust-policy.json

Reviewer: ```suggestion
$ cat <<EOF > /tmp/shared-vpc-role.json
```

{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Principal": {
                "AWS": "arn:aws:iam::<Account-ID>:root"  <1>
            },
            "Action": "sts:AssumeRole"
        }
    ]
}
EOF
----
+
--
<1> The principal will be scoped down later in this process after the *cluster-creating AWS account* user has created the necessary cluster roles. On creation, you must create a root user placeholder by using the *cluster-creator's AWS Account* ID as `arn:aws:iam::{Account}:root`.
--
+
. Create the IAM role:
+
[source, terminal]
----
$ aws iam create-role --role-name <role_name> \  <1>

>     --assume-role-policy-document file:///tmp/trust-policy.json

Reviewer: ```suggestion
    --assume-role-policy-document file:///tmp/shared-vpc-role.json
```

----
+
--
<1> Replace _<role_name>_ with the name of the role you want to create.
--
+

> . Attach the custom `Route53minimalPermissions` permissions policy:

Reviewer: ```suggestion
. Attach the custom `SharedVPCPolicy` permissions policy:
```

+
[source, terminal]
----
$ aws iam attach-role-policy --role-name <role_name> --policy-arn \  <1>

>     arn:aws:iam::<AWS_account_ID>:policy/Route53minimalPermissions  <2>

Reviewer: ```suggestion
    arn:aws:iam::<AWS_account_ID>:policy/SharedVPCPolicy  <2>
```

----
+
--
<1> Replace _<role_name>_ with the name of the role you have created.
<2> Replace
_<AWS_account_ID>_ with the *VPC-owning AWS account* ID.
--

> +

Reviewer: ```suggestion
```

> . Attach the AWS managed link:https://docs.aws.amazon.com/aws-managed-policy/latest/reference/ResourceGroupsandTagEditorFullAccess.html[`ResourceGroupsandTagEditorFullAccess`] permissions policy:

Reviewer: ```suggestion
```

> +

Reviewer: ```suggestion
```

> [source, terminal]

Reviewer: ```suggestion
```

> ----
> $ aws iam attach-role-policy --role-name <role_name> --policy-arn \
>     arn:aws:iam::aws:policy/ResourceGroupsandTagEditorFullAccess
> ----

Reviewer: ```suggestion
```
